util/types/time.go

```diff
@@ -167,6 +179,12 @@ type Time struct {
 	// TODO: Define a type for timestamp, remove its representation from here.
 	// TimeZone is valid when Type is mysql.Timestamp, it's meaningless for date/datetime.
 	TimeZone *gotime.Location
+	negative bool
+}
+
+// MaxMySQLTime returns Time with maximum mysql time type.
+func MaxMySQLTime(neg bool, fsp int) Time {
```

Comment:
MaxMySQLTime is 838:59:59 but not 838:59:59.999999?

Reply:
port from MySQL. :)